### PR TITLE
En_Floormas

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Floormas/z_en_floormas.c [hunk 1]
```diff
@@ -91,144 +141,1104 @@ static DamageTable D_808D38D4 = {
...
+}
+
+void func_808D09CC(EnFloormas* this) {
+    this->unk_18C = 10;
+    this->unk_2C8 = 0.55f;
+    this->unk_2CC = 0.82500005f;
+    this->unk_2C4 = 1.0f;
+    this->collider.base.colType = COLTYPE_HIT3;
+    this->unk_18E = 80;
+    this->actor.flags &= ~0x600;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Floormas/z_en_floormas.c:253]:
```suggestion
    this->actor.flags &= ~(0x400 | 0x200);
```

File: src/overlays/actors/ovl_En_Floormas/z_en_floormas.c [hunk 2]
```diff
@@ -91,144 +141,1104 @@ static DamageTable D_808D38D4 = {
...
+    this->actor.speedXZ = 0.0f;
+    if (sp36 > 0) {
+        SkelAnime_ChangeAnimTransitionStop(&this->skelAnime, &D_06002158, -3.0f);
+    } else {
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_06002158, -1.0f, SkelAnime_GetFrameCount(&D_06002158.common), 0.0f, 2,
+                             -3.0f);
+    }
+
+    if (this->actor.scale.x > 0.009f) {
+        this->unk_190 = sp36 * 0.06666667f;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Floormas/z_en_floormas.c:394]:
```suggestion
        this->unk_190 = sp36 * (1.0f / 60.0f);
```

File: src/overlays/actors/ovl_En_Floormas/z_en_floormas.c [hunk 3]
```diff
@@ -91,144 +141,1104 @@ static DamageTable D_808D38D4 = {
...
+    } else {
+        this->skelAnime.animPlaybackSpeed *= 2.0f;
+        this->unk_190 = sp36 * 0.13333334f;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Floormas/z_en_floormas.c:397]:
```suggestion
        this->unk_190 = sp36 * (2.0f / 15.0f);
```
I suspect this won't match, but something like `(2.000001f / 15.0f)` would be nice

File: src/overlays/actors/ovl_En_Floormas/z_en_floormas.c [hunk 4]
```diff
@@ -91,144 +141,1104 @@ static DamageTable D_808D38D4 = {
...
+}
+
+void func_808D1650(EnFloormas* this, GlobalContext* globalCtx) {
+    f32 temp_f0_2;
+
+    if (this->unk_18E != 0) {
+        this->unk_18E--;
+    }
+
+    Math_StepToF(&this->actor.speedXZ, 15.0f, SQ(this->actor.speedXZ) * 0.33333334f);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Floormas/z_en_floormas.c:486]:
```suggestion
    Math_StepToF(&this->actor.speedXZ, 15.0f, SQ(this->actor.speedXZ) * (1.0f / 3.0f));
```
Similar idea to the previous comment

File: src/overlays/actors/ovl_En_Floormas/z_en_floormas.c [hunk 5]
```diff
@@ -91,144 +141,1104 @@ static DamageTable D_808D38D4 = {
...
+
+void func_808D1458(EnFloormas* this, GlobalContext* globalCtx) {
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        func_808D161C(this);
+    }
+
+    this->actor.shape.rot.x += 0x140;
+    this->actor.world.pos.y += 10.0f;
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 3, 0xAAA);
+    Math_StepToS(&this->unk_192, 0x4B0, 0x64);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Floormas/z_en_floormas.c:450]:
```suggestion
    Math_StepToS(&this->unk_192, 1200, 100);
```
`this->unk_192` seems to be a pos

File: src/overlays/actors/ovl_En_Floormas/z_en_floormas.c [hunk 6]
```diff
@@ -91,144 +141,1104 @@ static DamageTable D_808D38D4 = {
...
+                func_808D2700(this);
+            } else {
+                func_808D0C14(this);
+                this->unk_194 = 50;
+            }
+        }
+    }
+
+    Math_ScaledStepToS(&this->actor.shape.rot.x, 0, 0x140);
+    Math_StepToS(&this->unk_192, -0x640, 0x64);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Floormas/z_en_floormas.c:563]:
```suggestion
    Math_StepToS(&this->unk_192, -1600, 100);
```

File: src/overlays/actors/ovl_En_Floormas/z_en_floormas.c [hunk 7]
```diff
@@ -91,144 +141,1104 @@ static DamageTable D_808D38D4 = {
...
+                            this->unk_2C8 = 0.55f;
+                            this->unk_18C = 0;
+                        } else if (this->actor.colChkInfo.damageEffect == 4) {
+                            this->unk_2C4 = 4.0f;
+                            this->unk_2C8 = 0.55f;
+                            this->unk_18C = 20;
+                            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_CLEAR_TAG,
+                                        this->collider.info.bumper.hitPos.x, this->collider.info.bumper.hitPos.y,
+                                        this->collider.info.bumper.hitPos.z, 0, 0, 0,
+                                        (this->actor.scale.x > 0.009f) ? 4 : 3);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Floormas/z_en_floormas.c:1084]:
```suggestion
                                        (this->actor.scale.x > 0.009f) ? CLEAR_TAG_LARGE_LIGHT_RAYS : CLEAR_TAG_SMALL_LIGHT_RAYS);
```

File: src/overlays/actors/ovl_En_Floormas/z_en_floormas.c [hunk 8]
```diff
@@ -91,144 +141,1104 @@ static DamageTable D_808D38D4 = {
...
+        SysMatrix_GetStateTranslationAndScaledX(1000.0f, &this->unk_2D0[9]);
+        SysMatrix_GetStateTranslationAndScaledX(-1000.0f, &this->unk_2D0[10]);
+        return;
+    }
+
+    if (limbIndex == 2) {
+        SysMatrix_StatePush();
+        SysMatrix_InsertTranslation(1600.0f, -700.0f, -1700.0f, MTXMODE_APPLY);
+        SysMatrix_InsertYRotation_f(1.0471976f, MTXMODE_APPLY);
+        SysMatrix_InsertZRotation_f(0.2617994f, MTXMODE_APPLY);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Floormas/z_en_floormas.c:1188]:
```suggestion
        SysMatrix_InsertYRotation_f(M_PI / 3, MTXMODE_APPLY);
        SysMatrix_InsertZRotation_f(M_PI / 12, MTXMODE_APPLY);
```

**Review by engineer124 — COMMENTED** [review]
Great work! Just a few small comments

File: src/overlays/actors/ovl_En_Floormas/z_en_floormas.c (hunk 2 above)
Comment (Kelebek1) [reply src/overlays/actors/ovl_En_Floormas/z_en_floormas.c:394]: This doesn't match.

File: src/overlays/actors/ovl_En_Floormas/z_en_floormas.c [hunk 9]
```diff
@@ -91,144 +141,1105 @@ static DamageTable D_808D38D4 = {
...
+
+    flags = this->actor.bgCheckFlags & 8;
+    if (flags) {
+        this->unk_190 = this->actor.wallYaw;
+        func_808D108C(this);
+        return;
+    }
+
+    if (this->actor.params == ENFLOORMAS_GET_7FFF_20) {
+        if (this->actor.parent->params == ENFLOORMAS_GET_7FFF_40) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Floormas/z_en_floormas.c:660]:
```suggestion
        func_808D108C(this);
    } else if (this->actor.params == ENFLOORMAS_GET_7FFF_20) {
        if (this->actor.parent->params == ENFLOORMAS_GET_7FFF_40) {
```

File: src/overlays/actors/ovl_En_Floormas/z_en_floormas.c [hunk 10]
```diff
@@ -91,144 +141,1105 @@ static DamageTable D_808D38D4 = {
...
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+
+    this->unk_192 = -1600;
+    params = ENFLOORMAS_GET_8000(&this->actor);
+    this->actor.params = ENFLOORMAS_GET_7FFF(&this->actor);
+
+    if (params) {
+        this->actor.flags |= 0x80;
+        this->actor.draw = func_808D3754;
+    }
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Floormas/z_en_floormas.c:180]:
```suggestion
    if (params != 0) {
        this->actor.flags |= 0x80;
        this->actor.draw = func_808D3754;
    }
```
I don't think params in this case is a bool

File: src/overlays/actors/ovl_En_Floormas/z_en_floormas.c [hunk 11]
```diff
@@ -91,144 +141,1105 @@ static DamageTable D_808D38D4 = {
...
+    this->actor.speedXZ = 0.0f;
+    if (sp36 > 0) {
+        SkelAnime_ChangeAnimTransitionStop(&this->skelAnime, &D_06002158, -3.0f);
+    } else {
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_06002158, -1.0f, SkelAnime_GetFrameCount(&D_06002158.common), 0.0f, 2,
+                             -3.0f);
+    }
+
+    if (this->actor.scale.x > 0.009f) {
+        this->unk_190 = sp36 * 0.06666667f;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Floormas/z_en_floormas.c:394]:
```suggestion
        this->unk_190 = sp36 * (1.0f / 15.0f)
```
I think it was suppose to be this

**Review by engineer124 — COMMENTED** [review]
Nice, just two follow-up comments

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**